### PR TITLE
Filter builder test fixes

### LINKED ISSUE #922: Update Filter Builders
- Add `time-dimension` filter-builder

### PR DESCRIPTION
**Pull request by holderb5**
Resolves #922 

## Description
Create working filter builders, and uniform tests for them, add in a date-dimension test
## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/tests/integration/components/filter-builders/date-dimension-test.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    await this.owner.lookup('service:navi-metadata').loadMetadata();
+  });
+
+  test('it renders', async function(this: TestContext, assert) {
+    await render(hbs`
+        <FilterBuilders::DateDimension
+          @filter={{this.filter}}
+          @isCollapsed={{false}} />
+        />`);
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/filter-builders/date-dimension-test.ts:35]: Can we extract this to `const TEMPLATE = hbs...` and change `isCollapsed=true` for the collapsed test

File: packages/reports/tests/integration/components/filter-builders/dimension-test.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    );
+    await this.owner.lookup('service:navi-metadata').loadMetadata();
+  });
+
+  test('it renders', async function(this: TestContext, assert) {
+    await render(hbs`
+    <FilterBuilders::Dimension
+      @filter={{this.filter}}
+      @isCollapsed={{false}} />
+    />`);
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/filter-builders/dimension-test.ts:34]: same here

File: packages/reports/tests/integration/components/filter-builders/metric-test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    this.set('filter', factory.createFilter('metric', 'bardOne', 'pageViews', {}, 'gt', [30]));
+    await this.owner.lookup('service:navi-metadata').loadMetadata();
+  });
+
+  test('displayName', async function(this: TestContext, assert) {
+    await render(
+      hbs`<FilterBuilders::Metric 
+        @filter={{this.filter}} 
+        @isCollapsed={{false}} 
+      />`
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/filter-builders/metric-test.ts:32]: same

File: packages/reports/addon/components/filter-builders/base.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    return this.supportedOperators.find(({ id }) => id === this.args.filter.operator);
+  }
+
+  @computed('supportedOperators', 'args.filter.{operator,metric,values.[]}')
+  get filter() {
+    const { values, validations } = this.args.filter;
+    const { selectedOperator: operator } = this;
+
+    return {
+      subject: this.args.filter,
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/filter-builders/base.ts:53]: Can we rename this to `fragment` to be more clear

File: packages/reports/addon/components/filter-builders/date-time.js [hunk 5]
```diff
@@ -151,7 +151,7 @@ export default class DateTimeFilterBuilder extends BaseFilterBuilderComponent {
   /**
    * @property {String} displayName - display name for the filter
    */
-  @readOnly('filter.subject.name') displayName;
+  @readOnly('filter.displayName') displayName;
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/date-time.js:154]: yup you're right

File: packages/reports/addon/components/filter-builders/date-time.js [hunk 6]
```diff
@@ -163,7 +163,7 @@ export default class DateTimeFilterBuilder extends BaseFilterBuilderComponent {
 
     return {
       subject: { name: `Date Time (${this.dateTimePeriodName})` },
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/date-time.js:165]: I dont think we're using dateTime anymore, we're using DateDimension

File: packages/reports/addon/components/filter-builders/metric.hbs [hunk 7]
```diff
@@ -34,12 +35,13 @@
       {{operator.name}}
     </PowerSelect>
     <span class="filter-builder__values">
-      {{component
-        this.filter.operator.valuesComponent
-        filter=this.filter
-        request=@request
-        onUpdateFilter=@onUpdateFilter
-      }}
+      {{#let this.selectedOperator.valuesComponent as |ValuesComponent|}}
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/metric.hbs:38]: Doesnt seem to effect the tests eitherway, and the reports page seems to work both ways

File: packages/reports/addon/components/filter-builders/date-time.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/date-time.js:154]: Yup

File: packages/reports/addon/components/filter-builders/dimension.hbs [hunk 8]
```diff
@@ -26,25 +27,9 @@
...
-        @onchange={{this.setField}}
-        as | filterField |
-      >
-        {{filterField}}
-      </PowerSelect>
-    {{/if}}
     <span class="filter-builder-dimension__values {{if this.showFields "filter-builder-dimension__values--short"}}">
       {{component
-        this.filter.operator.valuesComponent
+        this.selectedOperator.valuesComponent
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/dimension.hbs:32]: when doing this it gets an error for not being used, where do we use this?

[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/dimension.hbs:32]: this change breaks the filter value display

File: packages/reports/addon/components/filter-builders/metric.hbs (hunk 7 above)
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/metric.hbs:38]: Doesnt seem to make a difference eitherway

File: packages/reports/addon/components/filter-builders/base.ts (hunk 4 above)
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/base.ts:53]: the function?

File: packages/reports/addon/components/filter-builders/base.hbs [hunk 9]
```diff
@@ -27,7 +27,7 @@
     </PowerSelect>
     <span class="filter-builder__values">
       {{component
-        this.filter.operator.valuesComponent
+        this.selectedOperator.valuesComponent
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/base.hbs:30]: this causes a test to fail 
Screen Shot 2020-09-15 at 9 09 38 AM

File: packages/reports/addon/components/filter-builders/date-time.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/date-time.js:165]: We arent currently using date-time as far as I know. I believe we're only using date-dimension

File: packages/reports/tests/integration/components/filter-builders/collapsed-test.ts [hunk 10]
```diff
@@ -21,6 +21,7 @@ module('Integration | Component | filter-builders/collapsed', function(hooks) {
     this.set('displayName', 'Foo');
     this.set('filter', factory.createFilter('dimension', 'bardOne', 'Foo', {}, 'in', []));
     this.set('supportedOperators', [{ id: 'in', name: 'Equals', valuesComponent: 'mock/values-component' }]);
```
Comment (jkusa) [inline packages/reports/tests/integration/components/filter-builders/collapsed-test.ts:23]: Do we need this line now? 
```suggestion
```

File: packages/reports/tests/integration/components/filter-builders/date-dimension-test.ts [hunk 11]
```diff
@@ -12,6 +12,10 @@ import FilterFragment from 'navi-core/models/bard-request-v2/fragments/filter';
 interface TestContext extends Context {
   filter: FilterFragment;
 }
+const TEMPLATE = hbs`<FilterBuilders::DateDimension
+@filter={{this.filter}}
+@isCollapsed={{true}} />
+`;
```
Comment (jkusa) [inline packages/reports/tests/integration/components/filter-builders/date-dimension-test.ts:18]:
```suggestion
const TEMPLATE = hbs`
  <FilterBuilders::DateDimension
    @filter={{this.filter}}
    @isCollapsed={{true}} 
  />`;
```

File: packages/reports/tests/integration/components/filter-builders/dimension-test.ts [hunk 12]
```diff
@@ -12,6 +12,11 @@ import FilterFragment from 'navi-core/models/bard-request-v2/fragments/filter';
 interface TestContext extends Context {
   filter: FilterFragment;
 }
+const TEMPLATE = hbs`
+<FilterBuilders::Dimension
+  @filter={{this.filter}}
+  @isCollapsed={{this.isCollapsed}} />
+/>`;
```
Comment (jkusa) [inline packages/reports/tests/integration/components/filter-builders/dimension-test.ts:19]:
```suggestion
const TEMPLATE = hbs`
  <FilterBuilders::Dimension
    @filter={{this.filter}}
    @isCollapsed={{this.isCollapsed}}
  />`;
```

File: packages/reports/tests/integration/components/filter-builders/metric-test.ts [hunk 13]
```diff
@@ -13,6 +13,10 @@ import FragmentFactory from 'navi-core/services/fragment-factory';
 interface TestContext extends Context {
   filter: FilterFragment;
 }
+const TEMPLATE = hbs`<FilterBuilders::Metric 
+  @filter={{this.filter}} 
+  @isCollapsed={{this.isCollapsed}} 
+/>`;
```
Comment (jkusa) [inline packages/reports/tests/integration/components/filter-builders/metric-test.ts:19]:
```suggestion
const TEMPLATE = hbs`
  <FilterBuilders::Metric 
    @filter={{this.filter}} 
    @isCollapsed={{this.isCollapsed}} 
  />`;
```

File: packages/reports/tests/integration/components/filter-builders/collapsed-test.ts (hunk 10 above)
Comment (holderb5) [reply packages/reports/tests/integration/components/filter-builders/collapsed-test.ts:23]: Deleted, we dont have any references without the searching for the selected operator

File: packages/reports/addon/components/filter-builders/base.ts [hunk 14]
```diff
@@ -44,7 +44,7 @@ export default class BaseFilterBuilder extends Component<BaseFilterBuilderArgs>
     return this.supportedOperators.find(({ id }) => id === this.args.filter.operator);
   }
 
-  @computed('supportedOperators', 'args.filter.{operator,metric,values.[]}')
+  @computed('supportedOperator', 'args.filter.{operator,validations,values.[]}')
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/components/filter-builders/base.ts:47]:
```suggestion
  @computed('selectedOperator', 'args.filter.{validations,values.[]}')
```

File: packages/reports/addon/components/filter-builders/collapsed.hbs [hunk 15]
```diff
@@ -3,9 +3,9 @@
 {{#if @field}}
   <span class="filter-field--collapsed">({{@field}})</span>
 {{/if}}
```
[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/components/filter-builders/collapsed.hbs:5]: yes
```suggestion
```

File: packages/reports/tests/integration/components/filter-builders/date-dimension-test.ts [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+    this.set('isCollapsed', false);
+    await this.owner.lookup('service:navi-metadata').loadMetadata();
+  });
+
+  test('it renders', async function(this: TestContext, assert) {
+    await render(hbs`
+        <FilterBuilders::DateDimension
+          @filter={{this.filter}}
+          @isCollapsed={{false}} />
+        />`);
```
Comment (jkusa) [inline packages/reports/tests/integration/components/filter-builders/date-dimension-test.ts:39]:
```suggestion
    await render(TEMPLATE);
```

File: packages/reports/addon/components/filter-builders/metric.ts [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+
+import BaseFilterBuilderComponent from './base';
+
+export default class MetricFilterBuilderComponent extends BaseFilterBuilderComponent {
+  /**
+   * @property {Object[]} supportedOperators
+   */
+  supportedOperators = [
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/metric.ts:12]: Do you mean defining get supported Operators and have them return what we have defined as supportedOPerators?

[the remark this answers is not in the document]
Comment (kevinhinterlong) [reply packages/reports/addon/components/filter-builders/metric.ts:12]: yep

File: packages/reports/addon/components/filter-builders/collapsed.hbs (hunk 15 above)
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-builders/collapsed.hbs:5]: So do we not need the (desc) field part of the test anymore

[the remark this answers is not in the document]
Comment (jkusa) [reply packages/reports/addon/components/filter-builders/collapsed.hbs:5]: correct

File: packages/reports/addon/components/filter-builders/base.ts [hunk 18]
```diff
@@ -44,7 +44,7 @@ export default class BaseFilterBuilder extends Component<BaseFilterBuilderArgs>
     return this.supportedOperators.find(({ id }) => id === this.args.filter.operator);
   }
 
-  @computed('supportedOperator', 'args.filter.{operator,validations,values.[]}')
+  @computed('selectedOperator', 'args.filter.{operator,validations,values.[]}')
```
Comment (jkusa) [inline packages/reports/addon/components/filter-builders/base.ts:47]:
```suggestion
  @computed('selectedOperator', 'args.filter.{validations,values.[]}')
```

File: packages/reports/addon/components/filter-builders/base.ts (hunk 4 above)
Comment (jkusa) [reply packages/reports/addon/components/filter-builders/base.ts:53]: https://github.com/yahoo/navi/issues/1026

**Review by kevinhinterlong — APPROVED** [review]

**Review by jkusa — APPROVED** [review]

**Pull request merged**